samples/testing/basic/cases_test.go

```diff
@@ -0,0 +1,28 @@
+package basic
+
+import "testing"
+
+func TestCases(t *testing.T) {
+	// You can create many test cases succinctly with an anonymous struct array
+	testCases := []struct {
+		input, expected string
+	}{
+		{"Hello", "Hello M8"},
+		{"Goodbye", "Goodbye M8"},
+		{"💥💥💥", "💥💥💥 M8"},
+		{"", " M8"},
+	}
```

Comment:
Use map rather than slice for test case name and use T.Run (see https://dave.cheney.net/2019/05/07/prefer-table-driven-tests) e.g.:
```go
func TestSplit(t *testing.T) {
    tests := map[string]struct {
        input string
        sep   string
        want  []string
    }{
        "simple":       {input: "a/b/c", sep: "/", want: []string{"a", "b", "c"}},
        "wrong sep":    {input: "a/b/c", sep: ",", want: []string{"a/b/c"}},
        "no sep":       {input: "abc", sep: "/", want: []string{"abc"}},
        "trailing sep": {input: "a/b/c/", sep: "/", want: []string{"a", "b", "c"}},
    }

    for name, tc := range tests {
        tc := tc // get linter warning otherwise
        t.Run(name, func(t *testing.T) {
            got := Split(tc.input, tc.sep)
            if !reflect.DeepEqual(tc.want, got) {
                t.Fatalf("expected: %v, got: %v", tc.want, got)
            }
        })
    }
}
```
to get better output:
```
--- FAIL: TestSplit (0.00s)
    --- FAIL: TestSplit/trailing_sep (0.00s)
        split_test.go:25: expected: [a b c], got: [a b c ]
```

Reply:
Done